src/Tmds.Ssh/PrivateKeyCredential.cs

```diff
@@ -1,20 +1,94 @@
 // This file is part of Tmds.Ssh which is released under MIT.
 // See file LICENSE for full license details.
 
+using System.Security.Cryptography;
+
 namespace Tmds.Ssh;
 
-public sealed class PrivateKeyCredential : Credential
+public class PrivateKeyCredential : Credential
 {
-    internal string FilePath { get; }
+    internal string Identifier { get; }
+
+    private Func<CancellationToken, ValueTask<Key>> LoadKey { get; }
```

Comment:
`Key` seems to be `nullable` based on https://github.com/tmds/Tmds.Ssh/pull/223/files#diff-e8a24fa45fee86b727c97d8b80fe73f7113dcd748f6a7f871031b1e1d0d5f407R25-R30

Reply:
This `Key` is a `struct`, the nullable one is its nullable `PrivateKey` property that gets returned by the `LoadKeyAsync` (when `LoadKey` returns `default(Key)`).